### PR TITLE
Fixes and improvements to sygus satisfiable query generation

### PR DESCRIPTION
**Pull request by ajreynol**
Fixes issues related to the use of partial functions.  Also ensures we avoid duplicate queries.

### CODE REVIEW

File: src/theory/quantifiers/query_generator.cpp [hunk 1]
```diff
@@ -135,9 +141,14 @@ bool QueryGenerator::addTerm(Node n, std::ostream& out)
...
     {
       // take two random queries
-      std::shuffle(qsi.begin(), qsi.end(), Random::getRandom());
-      Node qy = nm->mkNode(AND, qsi[0], qsi[1]);
-      checkQuery(qy, i);
+      size_t rindex = Random::getRandom().pick(0, qsi.size() - 1);
+      size_t rindex2 = Random::getRandom().pick(0, qsi.size() - 1);
+      if (rindex == rindex2)
+      {
+        rindex2 = rindex + 1 == qsi.size() ? 0 : rindex + 1;
```
Comment (4tXJ7f) [inline src/theory/quantifiers/query_generator.cpp:148]: I guess we don't care that the distribution of pairs is not uniform anymore, right? Can we maybe add a comment about that?

**Review by 4tXJ7f — APPROVED** [review]
LGTM, it would just be good to add a comment regarding the non-uniformity of the query pairs we pick.

File: src/theory/quantifiers/query_generator.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/quantifiers/query_generator.cpp:148]: It's not a big deal, I'd prefer this than a while loop to avoid collisions.

Comment (ajreynol) [reply src/theory/quantifiers/query_generator.cpp]: Actually its simple to make it uniform, I've updated.

**Pull request merged**